### ISSUE TITLE
Error exporting to TFlite

### ISSUE THREAD

**Issue by anilmaddala** [tensorflow/tensorflow#23164]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04.1 LTS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: None
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): 1.10.1
- Python version: 
- Bazel version (if compiling from source):0.16.1
- GCC/Compiler version (if compiling from source):.3.0
- CUDA/cuDNN version: 9.1/7.1.2
- GPU model and memory: AWS EC2 g2.2 K520

**Describe the current behavior**
I want to reuse one of the classification checkpoints in mobilenet v2 as feature extractor in the object detection pipeline.

I started with a classification checkpoint mobilenet_v2_1.0_224 for object detection retraining.
I made modifications to ssd_mobilenet_v2_coco pipeline.config. Specifically changed `fine_tune_checkpoint_type` from `detection` to `classification` 
I uploaded the modified [pipeline.config](https://github.com/tensorflow/tensorflow/files/2502108/pipeline.config.txt) file.

I am able to retrain and I see the inference images in Tensorboard during the training process. The loss values converge.

The problem is when exporting to TFLite using `object_detection/export_tflite_ssd_graph.py` script.

Command to export to TFlite:
`python object_detection/export_tflite_ssd_graph.py 
                  --pipeline_config_path=$CONFIG_FILE 
--trained_checkpoint_prefix=$CHECKPOINT_PATH 
--output_directory=$OUTPUT_DIR 
--add_postprocessing_op=true`

Error:
InvalidArgumentError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

Assign requires shapes of both tensors to match. lhs shape= [3,3,256,24] rhs shape= [3,3,1280,24]
     [[Node: save/Assign_15 = Assign[T=DT_FLOAT, _class=["loc:@BoxPredictor_1/BoxEncodingPredictor/weights"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](BoxPredictor_1/BoxEncodingPredictor/weights, save/RestoreV2:15)]]

**Describe the expected behavior**
The retraining was successful, so I should get a tflite file successfully after exporting.

**Code to reproduce the issue**
N/A

**Other info / logs**
I am able to start with a detection checkpoint like ssd_mobilenet_v2_coco, retrrain it, export to TFLite and run inferece on Android.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by anilmaddala** [comment]
Updated the original issue comment.

**Comment by ymodak** [comment]
It looks like there is mismatch between model training configuration and model loading configuration.
Please refer [this](https://www.tensorflow.org/api_guides/python/meta_graph) and try using, Import a graph with preset devices as mentioned in the manual.

**Comment by tofulawrence** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
